### PR TITLE
concurrentqueue: init at 1.0.2.

### PR DESCRIPTION
**Pull request by invokesus**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/concurrentqueue/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+  src = fetchFromGitHub {
+    owner = "cameron314";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1wj5i2yvdapbmpjmn0pyk2c9v03194zckxa6n5g9havqlsby78gs";
+  };
+
+  # The makefile uses relative paths so setting makefile or makeFlags doesn't work.
+  preBuild = "cd build";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/concurrentqueue/default.nix:15]:
```suggestion
  preBuild = ''
    cd build
  '';
```

**Comment by invokesus** [comment]
Thanks for the review! Made the changes. :)

File: pkgs/development/libraries/concurrentqueue/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    sha256 = "1wj5i2yvdapbmpjmn0pyk2c9v03194zckxa6n5g9havqlsby78gs";
+  };
+
+  # The makefile uses relative paths so setting makefile or makeFlags doesn't work.
+  preBuild = ''
+    cd build
+  '';
+
+  #_FORTIFY_SOURCE requires compiling with optimization (-O)
+  NIX_CFLAGS_COMPILE = "-O";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/concurrentqueue/default.nix:20]: Do you know if you can set CFLAGS instead? NIX_CFLAGS_COMPILE should be the last resort.

File: pkgs/development/libraries/concurrentqueue/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  installPhase = ''
+    mkdir -p $out/{lib,include}
+    cp bin/libtbb.a $out/lib
+    cp ../*.h $out/include
+  '';
+
+  meta = with lib; {
+    description =
+      "A fast multi-producer, multi-consumer lock-free concurrent queue for C++11";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/concurrentqueue/default.nix:34]:
```suggestion
    description = "A fast multi-producer, multi-consumer lock-free concurrent queue for C++11";
```

Treewide will be easier if everything has a standard format.

**Comment by SuperSandro2000** [comment]
Closing due to inactivity from author.

**Pull request closed without merging**